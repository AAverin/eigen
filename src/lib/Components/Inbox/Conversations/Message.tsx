import * as React from "react"

import { BodyText, MetadataText, SmallHeadline } from "../Typography"

import Avatar from "./Avatar"

import styled from "styled-components/native"
import colors from "../../../../data/colors"

const VerticalLayout = styled.View`
  flex-direction: column
  flex: 1
`

const HorizontalLayout = styled.View`
  flex-direction: row
`

const Container = styled(HorizontalLayout)`
  alignSelf: stretch
  marginTop: 15
  marginBottom: 10
  marginLeft: 20
  marginRight: 20

`

const Header = styled(HorizontalLayout)`
  alignSelf: stretch
  marginBottom: 10
`

const TextContainer = styled(VerticalLayout)`
  marginLeft: 10
`

const SenderName = styled(SmallHeadline)`
  marginRight: 10
`

const ArtworkPreviewContainer = styled.View`
  marginBottom: 10
`

interface Props {
  message: {
    senderName: string
    time: string
    body: string
  }
  artworkPreview?: JSX.Element
}

export default class Message extends React.Component<Props, any> {
  render() {
    return (
      <Container>
        <Avatar isUser={true} senderName={this.props.message.senderName} />
        <TextContainer>
          <Header>
            <SenderName>{this.props.message.senderName}</SenderName>
            <MetadataText>{this.props.message.time}</MetadataText>
          </Header>
          {this.props.artworkPreview && <ArtworkPreviewContainer>{this.props.artworkPreview}</ArtworkPreviewContainer>}
          <BodyText>{this.props.message.body.split("\n\nAbout")[0]}</BodyText>
        </TextContainer>
      </Container>
    )
  }
}
