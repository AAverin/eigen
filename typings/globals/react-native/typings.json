{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
