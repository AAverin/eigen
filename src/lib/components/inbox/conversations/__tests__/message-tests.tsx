import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import Message from "../message"

it("looks correct when rendered", () => {
  const tree = renderer.create(<Message />).toJSON()
  expect(tree).toMatchSnapshot()
})
