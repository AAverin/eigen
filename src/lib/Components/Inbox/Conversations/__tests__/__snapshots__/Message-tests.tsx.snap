// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks correct when rendered 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Array [
        Object {
          "alignSelf": "stretch",
          "marginBottom": 10,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 15,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "borderRadius": 15,
          "height": 30,
          "width": 30,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Avant Garde Gothic ITCW01Dm",
            "fontSize": 10,
            "marginTop": 8,
          },
          undefined,
        ]
      }
      user={true}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        Array [
          Object {
            "marginLeft": 10,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "alignSelf": "stretch",
              "marginBottom": 10,
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 11,
              "textAlign": "left",
            },
            Array [
              Object {
                "marginRight": 10,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Avant Garde Gothic ITCW01Dm",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#cccccc",
              "fontSize": 11,
              "textAlign": "left",
            },
            Object {},
            Object {
              "fontFamily": "Avant Garde Gothic ITCW01Dm",
            },
          ]
        }
      >
        30 MINUTES
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 16,
            "textAlign": "left",
          },
          false,
          Object {},
          Object {
            "fontFamily": "AGaramondPro-Regular",
          },
        ]
      }
    >
      Hi, I'm interested in purchasing this work. Could you please provide more information about the piece, including price?
    </Text>
  </View>
</View>
`;
